src/twisted/trial/test/test_loader.py

```diff
@@ -66,6 +73,30 @@ def test_findFileWithImportError(self):
         )
 
 
+def loosely_resembles(module: ModuleType) -> Matcher[ModuleType]:
+    """
+    Match a module with a L{ModuleSpec} like that of the given module.
+
+    @return: A matcher for a module spec that has the same name and origin as
+        the given module spec, though the origin may be structurally inequal
+        as long as it is semantically equal.
+    """
+    expected = module.__spec__
+    if expected is None:
+        match_spec: Matcher[Optional[object]] = is_(None)
```

Comment:
Would it be possible to hit this with a synthetic module object?  Or maybe just `assert` it out if we don't need to handle this case yet?